src/libraries/Microsoft.PowerFx.Core/Public/Marshal/TableMarshallerProvider.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Public.Values;
+
+namespace Microsoft.PowerFx.Core
+{
+    /// <summary>
+    /// Marshal .Net classes (with fields). This supports strong typing and lazy marshalling. 
+    /// Handles any IEnumerable (including arrays).
+    /// </summary>
+    public class TableMarshallerProvider : ITypeMashallerProvider
+    {
+        /// <inheritdoc/>
+        public bool TryGetMarshaller(Type type, TypeMarshallerCache cache, int maxDepth, out ITypeMarshaller marshaler)
+        {
+            if (!IsIEnumerableT(type, out var et))
+            {
+                marshaler = null;
+                return false;
+            }
+
+            var tm = cache.GetMarshaller(et, maxDepth);
+             
+            if (tm.Type is RecordType recordType)
+            {
+                // Array of records 
+            }       
+            else
+            {
+                // Single Column table. Wrap the scalar in a record. 
```

Comment:
>scalar [](http://example.com/codeflow?start=49&length=6)

It may not be a scalar - it can be a table instead (e.g., [[1,2],[3,4,5]]). wrapping it in {Value:*} is still correct, it's just the comment that needs to be updated.